src/TimeModeling/Types/judiWeights.jl

```diff
@@ -42,7 +42,7 @@ Construct vector cell array of weights. The `weights` keyword\\
 can also be a single (non-cell) array, in which case the weights are the same for all source positions:
     judiWeights(weights; nsrc=1)
 """
-function judiWeights(weights::Array{T, N}; nsrc=1, vDT::DataType=Float32) where {T, N}
+function judiWeights(weights::Array{T, N}; nsrc=1, vDT::DataType=Float32) where {T<:Real, N}
```

Comment:
Why only judiWeights?

Reply:
Wavefield can be complex if fft applied for example.

Reply:
What if judiVector

Reply:
We don't support complex valued JUDI vector. For the wavefield we do because (potentially) because of on-the-fly DFT that produces complex values and is supported by the propagators, we don't have any support for complex valued data operations.

Reply:
Then why not adding `T<:Real` for example https://github.com/slimgroup/JUDI.jl/blob/c4a818d27c728b4833e406ac1a0113e693ff17ea/src/TimeModeling/Types/judiVector.jl#L86

Reply:
don't think it really matters

Reply:
ok